src/dsp_tools/commands/validate_data/process_validation_report/get_user_validation_message.py

```diff
@@ -122,56 +122,72 @@ def _group_problems_by_resource(problems: list[InputProblem]) -> dict[str, list[
         else:
             violation_body = _get_problem_print_message(sorted_problems.unique_violations)
+            violation_df = None
         violation_header = (
             f"During the validation of the data {len(sorted_problems.unique_violations)} errors were found. "
             f"Until they are resolved an xmlupload is not possible."
         )
-        violation_message = MessageStrings(violation_header, violation_body)
+        violation_message = MessageComponents(violation_header, violation_body, violation_df)
     if sorted_problems.user_warnings:
-        if len(sorted_problems.user_warnings) > 50:
-            warning_body = _save_problem_info_as_csv(sorted_problems.user_warnings, file_path, "warnings")
+        if too_many_to_print:
+            warning_body = None
+            warning_df = _get_message_df(sorted_problems.user_warnings)
         else:
             warning_body = _get_problem_print_message(sorted_problems.user_warnings)
+            warning_df = None
         warning_header = (
             f"During the validation of the data {len(sorted_problems.user_warnings)} "
             f"problems were found. Warnings are allowed on test servers. "
             f"Please note that an xmlupload on a prod sever will fail."
         )
-        warning_message = MessageStrings(warning_header, warning_body)
+        warning_message = MessageComponents(warning_header, warning_body, warning_df)
     if sorted_problems.user_info:
-        if len(sorted_problems.user_info) > 50:
-            info_body = _save_problem_info_as_csv(sorted_problems.user_info, file_path, "info")
+        if too_many_to_print:
+            info_body = None
+            info_df = _get_message_df(sorted_problems.user_info)
         else:
             info_body = _get_problem_print_message(sorted_problems.user_info)
+            info_df = None
         info_header = (
             f"During the validation of the data {len(sorted_problems.user_info)} "
             f"potential problems were found. They will not impede an xmlupload."
         )
-        info_message = MessageStrings(info_header, info_body)
+        info_message = MessageComponents(info_header, info_body, info_df)
     if sorted_problems.unexpected_shacl_validation_components:
         unexpected_header = "The following unknown violation types were found!"
         unexpected_body = LIST_SEPARATOR + LIST_SEPARATOR.join(sorted_problems.unexpected_shacl_validation_components)
-        unexpected_violations = MessageStrings(unexpected_header, unexpected_body)
+        unexpected_violations = MessageComponents(unexpected_header, unexpected_body, None)
     return UserPrintMessages(violation_message, warning_message, info_message, unexpected_violations)
 
 
+def _are_there_too_many_to_print(sorted_problems: SortedProblems, severity: ValidationSeverity) -> bool:
+    number_of_problems = len(sorted_problems.unique_violations)
+    if severity.value <= 2:
```

Comment:
If you compare enums this way, you loose a big part of their benefits. This way it would be much better understandable:

```suggestion
    if severity <= ValidationSeverity.WARNING:
```